sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ext/opentelemetry_span/__init__.py

```diff
@@ -228,7 +228,7 @@ def __enter__(self) -> "OpenTelemetrySpan":
     def __exit__(self, exception_type, exception_value, traceback) -> None:
         # Finish the span.
         if exception_type:
-            self.add_attribute(_ERROR_SPAN_ATTRIBUTE, str(exception_type.__name__))
+            self.add_attribute(_ERROR_SPAN_ATTRIBUTE, f"{exception_type.__module__}.{exception_type.__qualname__}")
```

Comment:
Actually, what about just using `repr`?

Reply:
E.g. 

```
>>> repr(Exception)
"<class 'Exception'>"
```

Reply:
Could be an option, however, I wonder if the formatting might deviate a bit too much from the expectations laid out by the OTel [semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/error.md). The error type is conveyed either way, but I do think it looks cleaner without the angle brackets and quotation marks in the value.

- `azure.core.exceptions.ClientAuthenticationError` vs `<class 'azure.core.exceptions.ClientAuthenticationError'>`
- `builtins.ValueError` vs `<class 'ValueError'>`

@lmolkova WDYT?

Reply:
We expect users to use it in their queries, dashboards, etc with something like this
`spans | where error_type="azure.core.exceptions.ClientAuthenticationError" | summarize count() by bin(timestamp, 1m)`

and `azure.core.exceptions.ClientAuthenticationError` looks more friendly than `<class 'azure.core.exceptions.ClientAuthenticationError'>`.

So I'd prefer `azure.core.exceptions.ClientAuthenticationError` but as @pvaneck mentioned, both forms are allowed.

Reply:
Will go with what upstream OTel Python is leaning towards using when logging events (based on discussion in https://github.com/open-telemetry/opentelemetry-python/issues/3824). So, we will prepend `__module__` if not a builtin exception.